### PR TITLE
Fix conversion to zero for E4M3FNUZ and E5M2FNUZ

### PR DESCRIPTION
**Pull request by xadupre**
### Description
-0.00001 must be converted into 0 and not negative zero which does not exist for types Float8E4M3FNUZ and Float8E5M2UZ.
See https://github.com/microsoft/onnxruntime/pull/18473 for further details.

### CODE REVIEW

File: docs/Operators.md [hunk 1]
```diff
@@ -3520,40 +3520,43 @@ for from_type, to_type in test_cases:
                 "INF",
                 "+INF",
                 "-INF",
+                "-0.0000001",
```
Comment (justinchuby) [inline docs/Operators.md:3523]: Just clarifying - is it just `-0.0000001` or do we do the same for `-0.1`, `-0.4` etc?

**Review by justinchuby — APPROVED** [review]

File: docs/Operators.md (hunk 1 above)
Comment (xadupre) [reply docs/Operators.md:3523]: -0.0000001 is rounded to 0 but before this fix, it was -0->nan and not +0 which was wrong

Comment (justinchuby) [reply docs/Operators.md:3523]: should any other negative values be rounded to +0 too?

Comment (gramalingam) [reply docs/Operators.md:3523]: Not `-0.1`, but some other values, yes ... that's my understanding. The [documentation](https://onnx.ai/onnx/technical/float8.html#e4m3fnuz-and-e5m2fnuz) mentions the `Min` value representable in the format ... eg., 2^-17 for E5M2 variant. I guess values in the range [-Min, 0] will be rounded to 0? Xavier can clarify.

**Review by gramalingam — APPROVED** [review]

**Pull request merged**